src/Form/DataTransformer/ArrayToModelTransformer.php

```diff
@@ -30,35 +32,57 @@ class ArrayToModelTransformer implements DataTransformerInterface
 
     /**
      * @var string
+     *
+     * @phpstan-var class-string<T>
      */
     protected $className;
 
     /**
      * @param string $className
+     *
+     * @phpstan-param class-string<T> $className
      */
     public function __construct(ModelManagerInterface $modelManager, $className)
     {
         $this->modelManager = $modelManager;
         $this->className = $className;
     }
 
-    public function reverseTransform($array)
+    /**
+     * @param object|array[]|null $value
```

Comment:
Are you sure it can be an array of array ?

Reply:
Good catch!